### ISSUE TITLE
Spike: Efficient Queries with Dataloaders

### ISSUE THREAD

**Issue by valentinyanakiev** [cherrytwist/Server#2602]
## Description

As an Alkemio user, I would like to have a performant application.

## Acceptance criteria

- [ ] graphql-efficient-sql integrated with dataloaders

## Additional Context

Timebox to half a day

**Comment by hero101** [comment]
@valentinyanakiev @techsmyth 

- I suggest rescheduling this piece of work for another time and closing this issue.
The current implementation is hard to go around and needs to be heavily reworked in order for fields to be passed to the batch function.
- It all boils down to passing the fields from an argument in the resolver to the batch function. The current implementation provides a way to pass the fields in question through the decorator, but since you cannot pass one argument to another argument in the same function definition (shown in the code below) - this is impossible.

```typescript
ResolveField('profile', () => IProfile, {
    nullable: false,
    description: 'The Profile for this User.',
  })
  @Profiling.api
  async profile(
    @Parent() user: User,
    @Info(ResolveInfoPipe) fields: Array<keyof IProfile>,
    @Loader(ProfileLoaderCreator, { parentClassRef: User, fields })
    loader: ILoader<IProfile>
  ): Promise<IProfile> {
    return loader.load(user.id);
  }
```
You are supposed to pass "fields" from argument 2 to argument 3, in the options of the decorator, which again - is impossible. I haven't found any workaround for this implementation.

- Another way of passing fields from the resolver to the batch function is passing them through the `load` method with a typed object along the lines of `{id: user.id, fields }`. This means that at the end the batch function will work with an array of objects all having different `id` but the same `fields`. This doesn't feel natural and it's not designed for this use-case, but is the only way to go that I have found.
- This approach needs a big time investment because all the types need to be changed, the designed utils need to be changed and all the loaders have to use the new implementation. 
- Having two approaches in parallel is also an option but will require the same effort.
- The performance benefit of this is measured to be around 10% per loader.

For these reasons, I suggest this to be addressed at a later stage with another issue.
All the code and findings are pushed to server-2602. The code is not buildable and not working.

**Comment by valentinyanakiev** [comment]
We are not going to pursue a gain of ~10% for a sizeable amount of effort.
